astropy/modeling/mappings.py

```diff
@@ -55,17 +59,20 @@ def inverse(self):
                 "Mappings such as {0} that drop one or more of their inputs "
                 "are not invertible at this time.".format(self.mapping))
 
-        return self.__class__(mapping)
+        inv = self.__class__(mapping)
+        inv._inputs = self._outputs
+        inv._outputs = self._inputs
+        return inv
 
 
 class Identity(Mapping):
-    def __init__(self, n_inputs):
-        mapping = tuple(range(n_inputs))
+    def __init__(self, n_outputs):
+        mapping = tuple(range(n_outputs))
```

Comment:
In other words, I think that the current behavior of `Identity` is a side-effect of it being a subclass of `Mapping`, which (deliberately) had this behavior.  But `Identity(1)` should have the same effect as `Mapping((0,), n_inputs=1)` which would make the number of inputs strict.